csharp/lib/src/lib.rs

```diff
@@ -153,7 +162,11 @@ pub extern "C" fn get(client_ptr: *const c_void, callback_index: usize, key: *co
             match result {
                 Ok(None) => (client.success_callback)(callback_index, std::ptr::null()),
                 Ok(Some(c_str)) => (client.success_callback)(callback_index, c_str.as_ptr()),
-                Err(_) => (client.failure_callback)(callback_index), // TODO - report errors
+                Err(err) => {
+                    logger_core::log_debug("command error", format!("callback {}, error {}, kind {:?}, code {:?}, category {:?}, detail {:?}", callback_index, err, err.kind(), err.code(), err.category(), err.detail()));
+                    let c_err_str = CString::new(error_message(&err)).expect("CString::new failed");
```

Comment:
You'll need to free the CString later, like I did in my Golang connection PR with `free_error`.